crates/noirc_evaluator/src/ssa_refactor/ir/function.rs

```diff
@@ -20,6 +23,34 @@ pub(crate) struct Function {
     entry_block: BasicBlockId,
 }
 
+impl Function {
+    /// Get an iterator over the ids of the basic blocks within the function.
+    ///
+    /// The ids are iterated in no meaningful order.
+    pub(crate) fn basic_block_ids_iter(&self) -> BasicBlockIdsIter {
+        BasicBlockIdsIter(self.basic_blocks.keys())
+    }
+}
+
+impl Index<BasicBlockId> for Function {
+    type Output = BasicBlock;
+    /// Get a function's basic block for the given id.
+    fn index<'a>(&'a self, id: BasicBlockId) -> &'a BasicBlock {
+        &self.basic_blocks[&id]
+    }
+}
+
+/// An iterator over a function's basic block ids. The iterator type is `BasicBlockId`.
+pub(crate) struct BasicBlockIdsIter<'a>(hash_map::Keys<'a, BasicBlockId, BasicBlock>);
+
+impl<'a> Iterator for BasicBlockIdsIter<'a> {
+    type Item = BasicBlockId;
+
+    fn next(&mut self) -> Option<BasicBlockId> {
+        self.0.next().map(|k| *k)
+    }
+}
```

Comment:
We can avoid adding a new iterator here if we change the function's `basic_blocks` to be a `DenseMap<BasicBlock>`, and add an iter function to that instead.

The iterator there should probably iterate over the id and value, so we can still have the helper function here for mapping the iterator to only the ids.